app/upgrades/v17/upgrades.go

```diff
@@ -0,0 +1,300 @@
+// This will also register the relevant community pool ICA addresses
+func RegisterCommunityPoolAddresses(ctx sdk.Context, k stakeibckeeper.Keeper) error {
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+		chainId := hostZone.ChainId
+
+		// Create and store a new community pool stake and redeem module address
+		stakeHoldingAddress := stakeibctypes.NewHostZoneModuleAddress(
+			chainId,
+			stakeibckeeper.CommunityPoolStakeHoldingAddressKey,
+		)
+		redeemHoldingAddress := stakeibctypes.NewHostZoneModuleAddress(
+			chainId,
+			stakeibckeeper.CommunityPoolRedeemHoldingAddressKey,
+		)
+
+		if err := utils.CreateModuleAccount(ctx, k.AccountKeeper, stakeHoldingAddress); err != nil {
+			return errorsmod.Wrapf(err, "unable to create community pool stake account for host zone %s", chainId)
+		}
+		if err := utils.CreateModuleAccount(ctx, k.AccountKeeper, redeemHoldingAddress); err != nil {
+			return errorsmod.Wrapf(err, "unable to create community pool redeem account for host zone %s", chainId)
+		}
+
+		hostZone.CommunityPoolStakeHoldingAddress = stakeHoldingAddress.String()
+		hostZone.CommunityPoolRedeemHoldingAddress = redeemHoldingAddress.String()
+
+		k.SetHostZone(ctx, hostZone)
+
+		// Register the deposit and return ICA addresses
+		// (these will get set in the OnChanAck callback)
+		// create community pool deposit account
+		connectionId := hostZone.ConnectionId
+		connectionEnd, found := k.IBCKeeper.ConnectionKeeper.GetConnection(ctx, connectionId)
+		if !found {
+			return errorsmod.Wrapf(connectiontypes.ErrConnectionNotFound, "connection %s not found", connectionId)
+		}
+		counterpartyConnectionId := connectionEnd.Counterparty.ConnectionId
+
+		appVersion := string(icatypes.ModuleCdc.MustMarshalJSON(&icatypes.Metadata{
+			Version:                icatypes.Version,
+			ControllerConnectionId: connectionId,
+			HostConnectionId:       counterpartyConnectionId,
+			Encoding:               icatypes.EncodingProtobuf,
+			TxType:                 icatypes.TxTypeSDKMultiMsg,
+		}))
+
+		depositAccount := stakeibctypes.FormatHostZoneICAOwner(chainId, stakeibctypes.ICAAccountType_COMMUNITY_POOL_DEPOSIT)
+		if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, connectionId, depositAccount, appVersion); err != nil {
+			return errorsmod.Wrapf(stakeibctypes.ErrFailedToRegisterHostZone, "failed to register community pool deposit ICA")
+		}
+
+		returnAccount := stakeibctypes.FormatHostZoneICAOwner(chainId, stakeibctypes.ICAAccountType_COMMUNITY_POOL_RETURN)
+		if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, connectionId, returnAccount, appVersion); err != nil {
+			return errorsmod.Wrapf(stakeibctypes.ErrFailedToRegisterHostZone, "failed to register community pool return ICA")
+		}
+	}
+
+	return nil
+}
+
+// Deletes all stale queries
+func DeleteAllStaleQueries(ctx sdk.Context, k icqkeeper.Keeper) {
+	for _, query := range k.AllQueries(ctx) {
+		if query.CallbackId == stakeibckeeper.ICQCallbackID_Delegation {
```

Comment:
Why is this the only query type that we want to flush on upgrade?  Would it be a good idea to clear any slash queries since we are explicitly resetting the slashQueryInProgress flag as well?  What about more mundane things like the pool exchange rate query or queries for amounts in different ICAs, will those not have any potential interaction effects with things being modified by the upgrade handlers?

Reply:
These are the slash queries! And the other queries you mentioned aren't live yet so they wouldn't be pending at the time that the upgrade is run